docs_src/design/adr/core/0019-EdgeX-CLI-V2.md

```diff
@@ -0,0 +1,104 @@
+# EdgeX-CLI V2 Design
+ 
+
+## Status
+
+**Draft**
+
+## Context
+This ADR presents a technical plan for creation of a 2.0 version of edgex-cli which supports the new V2 REST APIs developed as part of the Ireland release of EdgeX.
+
+## Existing Behavior
+The latest version of edgex-cli  (1.0.1) only supports the V1 REST APIs and thus cannot be used with V2 releases of EdgeX.
+
+As the edgex-cli was developed organically over time, the current implementation has a number of bugs mostly involving a lack of consistent behavior, especially with respect to formatting of output.
+
+Other issues with the existing client include:
+
+- lack of tab completion
+- default output of commands is too verbose
+- verbose output sometime prevents use of jq 
+- static configuration file required (i.e. no registry support)
+- project hierarchy not conforming to best practice guidelines 
+
+## History
+The original Hanoi V1 client was created by a team at VMWare which is no longer participating in the project. 
+Canonical will lead the development of the Ireland/Jakarta V2 client.
+
+## Decision
+
+1. Use standardized command-line args/flags
+
+| Argument/Flag      | Description |
+| ----------- | ----------- |
+| `-d`, `--debug`      | show additional output for debugging purposes (e.g. REST URL, request JSON, …). This command-line arg will replace -v, --verbose and will no longer trigger output of the response JSON (see -j, --json).       |
+| `-j`, `--json`   | output the raw JSON response returned by the EdgeX REST API and *nothing* else. This output mode is used for script-based usage of the client.    |
+| `--version`   | output the version of the client and if available, the version of EdgeX installed on the system (using the version of the core data service)   |
```

Comment:
```suggestion
| `--version`   | output the version of the client and if available, the version of EdgeX installed on the system (using the version of the core metadata service)   |
```

Reply:
Core Data is technically now optional...

Reply:
changed to refer to metadata (which is what the implementation already does).